tests/test_ml_kem.py

```diff
@@ -90,80 +58,89 @@ def test_xof_failure(self):
     def test_prf_failure(self):
         self.assertRaises(ValueError, lambda: ML_KEM_512._prf(2, b"1", b"2"))
 
+    def test_decaps_ct_type_check_failure(self):
+        """
+        Send a ciphertext of the wrong length
+        """
+        ek, dk = ML_KEM_512.keygen()
+        K, c = ML_KEM_512.encaps(ek)
+        self.assertRaises(ValueError, lambda: ML_KEM_512.decaps(dk, b"1"))
+
+    def test_decaps_dk_type_check_failure(self):
+        """
+        Send a ciphertext of the wrong length
+        """
+        ek, dk = ML_KEM_512.keygen()
+        K, c = ML_KEM_512.encaps(ek)
+        self.assertRaises(ValueError, lambda: ML_KEM_512.decaps(b"1", c))
+
+    def test_decaps_hash_check_failure(self):
+        """
+        Send a ciphertext of the wrong length
+        """
+        ek, dk = ML_KEM_512.keygen()
+        K, c = ML_KEM_512.encaps(ek)
+        dk_bad = b"0" * len(dk)
+        self.assertRaises(ValueError, lambda: ML_KEM_512.decaps(dk_bad, c))
+
+
+class TestML_KEM_KAT(unittest.TestCase):
+    """
+    Test ML_KEM levels for internal
+    consistency by generating key pairs
+    and shared secrets.
+    """
+
+    def generic_keygen_kat(self, ML_KEM, index):
+        with open("assets/ML-KEM-keyGen-FIPS203/internalProjection.json") as f:
+            data = json.load(f)
+        kat_data = data["testGroups"][index]["tests"]
+
+        for test in kat_data:
```

Comment:
this again runs all the KATs as a single test... :disappointed:

Reply:
Yeah but there's only 70 in total so it doesn't really matter. I'm happy the test being "everything matches the test"